http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/ChannelPipelineInitializer.java

```diff
@@ -109,7 +109,14 @@ public void channelCreated(Channel ch) {
         }
 
         pipeline.addLast(FutureCancelHandler.getInstance());
-        pipeline.addLast(UnusedChannelExceptionHandler.getInstance());
+
+        // Only add it for h1 channel because it does not apply to
+        // h2 connection channel. It will be attached
+        // to stream channels when they are created.
+        if (protocol == Protocol.HTTP1_1) {
+            pipeline.addLast(UnusedChannelExceptionHandler.getInstance());
```

Comment:
What happens when an exception is fired on a parent H2 channel after this change?

Reply:
`ReleaseOnExceptionHandler` will take care of it.

https://github.com/aws/aws-sdk-java-v2/blob/138def267fc8814418e5b7c8ab80f038b9d40641/http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/http2/Http2MultiplexedChannelPool.java#L380-L396